gui/launcher.cpp

```diff
@@ -393,6 +394,7 @@ void LauncherDialog::massAddGame() {
 	MessageDialog alert(_("Do you really want to run the mass game detector? "
 						  "This could potentially add a huge number of games."), _("Yes"), _("No"));
 	if (alert.runModal() == GUI::kMessageOK && _browser->runModal() > 0) {
+		MD5Man.clear();
```

Comment:
You need to run it not only during massAdd, but during the conventional detection too, since it always goes through all the engines.

I would say, the more proper place for this would be somewhere around `DetectionResults EngineManager::detectGames(const Common::FSList &fslist)`

Reply:
Thank you for the suggestions. I've force pushed the suggestion regarding `Common::String::format`.
As for the second suggestion,
I had put `MD5Man.clear()` inside of `DetectionResults EngineManager::detectGames(const Common::FSList &fslist)` before, but noticed that the amount of cache hits decreased when the engine had `directoryGlobs` defined. This is most likely because the file (located in a subfolder defined in directoryGlobs) is added to `allFiles` and has their md5 computed, but the cache is still cleared between folder jumps inside `void MassAddDialog::handleTickle()`.
To solve this I thought using the entire path as the hash key would be a good choice since there is (supposedly) no chance of the wrong md5 cache being used that way.
I could be understanding the system incorrectly and this solution could be more detrimental than I thought.

Reply:
Force pushed the second suggestion since not clearing the cache may have unwanted side effects. Thank you for your comments.